src/lib/alloc.c

```diff
@@ -259,6 +296,9 @@ static void *alloc_cont_blocks(struct mm_heap *heap, int level,
 	/* we found enough space, let's allocate it */
 	map->free_count -= count;
 	ptr = (void *)(map->base + start * map->block_size);
+
+	ptr = align_buffer_ptr(heap, alignment, ptr);
```

Comment:
I think at this point it is too late to apply alignment.

Reply:
How can i align it earlier? 
We just got the first free pointer a line before. 
What do You have in mind here ?